src/server.c

```diff
@@ -214,19 +214,14 @@ mstime_t commandTimeSnapshot(void) {
      * only the first time it is accessed and not in the middle of the
      * script execution, making propagation to slaves / AOF consistent.
      * See issue #1525 on Github for more information. */
-    if (server.script_caller) {
-        return scriptTimeSnapshot();
```

Comment:
please update the comments

Reply:
i left the comment on purpose (they're both still valid).
it looks odd though, and i can rephrase that whole thing into one big block comment rather than two medium ones, but i thought it's better to leave it unchanged.
so you see anything that's not valid anymore? or do you think it's too ugly to keep them as they are?

Reply:
i think it's a long explanation for saying "in case the command looks up more than one key, we need to use the cached time to avoid indeterministic lazy-expiry" or something alike

Reply:
ok, i'll rephrase it.

Reply:
seems we cannot remove the script time snapshot, the command snapshot is updated not only when execution in `call()` but also when script is busy call `processEventsWhileBlocked`:
```
scriptInterrupt()
-->
    processEventsWhileBlocked();
-->
        updateCachedTime(0);
```
the `execution_nesting` cannot protect the script's time snapshot.

Reply:
l looked at the one in `call` which does
```c
    if (server.fixed_time_expire++ == 0) {
        updateCachedTimeWithUs(0,call_timer);
    }
```
i now realize we have others...
i'll try to come up with another solution, maybe remember some command time snapshot in blockingOperationStarts and restore it later?
i.e. anything that's not specific to script and script caller, and is rather about nesting level.

Reply:
solved by adding a new `server.cmd_time_snapshot`, please take a look